[1.21] Add most remaining schema documentation (#1567)
Shadows-of-Fire/Apotheosis: schema/affix/AffixDefinition.md

# Description

> The AffixDefinition defines the core properties of an Affix, including its type, exclusivity rules, and tier-specific weights. This is used in conjunction with specific Affix implementations to create a complete Affix configuration.

Author: ```suggestion
An affix definition defines the core properties of an Affix, including its type, exclusivity rules, and weights. Each Affix will always specify an affix definition in its json schema.
```

# Dependencies
This object references the following objects:
1. [AffixType](./AffixType.md)
2. [TieredWeights](../tier/TieredWeights.md)

# Schema
```js
{
    "affix_type": AffixType,             // [Mandatory] || The type of affix (STAT, BASIC_EFFECT, or ABILITY)

>     "exclusive_set": [                   // [Mandatory] || Set of affix IDs that cannot be applied alongside this affix
>         string
>     ],

Author: ```suggestion
    "exclusive_set": [                   // [Mandatory] || Set of affix IDs that cannot be applied alongside this affix
        "string"
    ],
```

    "weights": TieredWeights             // [Mandatory] || Tier-specific weights determining how often this affix appears
}
```

# Examples

## Basic Definition

> A basic affix definition for a stat-based affix that appears in all world tiers.

Author: ```suggestion
A basic affix definition for a stat-type affix that appears in all world tiers.
```
